src/invoice.rs

```diff
@@ -112,6 +111,24 @@ pub enum InvoiceParseError {
     IfaceName(InvalidIdent),
 }
 
+impl std::fmt::Display for RgbInvoice {
+    fn fmt(&self, f: &mut std::fmt::Formatter) -> std::fmt::Result {
+        // TODO: Provide support for optionals & query
+        write!(
+            f,
+            "{}{}/{}/{}{}",
+            self.transport,
+            self.contract,
+            self.iface,
+            match self.value {
+                TypedState::Amount(amt) => format!("{amt}+"),
+                _ => String::new(),
+            },
```

Comment:
We need to add `Dsiplay` impl for `TypedState` (if it's not there) and use it here.

Reply:
`Display` for `TypedState` is missing, yes. But I'm not sure how you want to represent values that are not of `TypedState::Amount` type, any suggestion?

Reply:
I did Display for TypedState. Just rebase on this commit: https://github.com/RGB-WG/rgb-wallet/pull/32/commits/ce92118465fcce28be1405776ac34a974c679786